storage/innobase/srv/srv0start.cc

```diff
@@ -1491,6 +1491,67 @@ srv_prepare_to_delete_redo_log_files(
 	return(flushed_lsn);
 }
 
+/** Enable encryption of system tablespace if requested. At
+startup load the encryption information from first datafile
+to tablespace object
+@return DB_SUCCESS on succes, others on failure */
+static
+dberr_t
+srv_sys_enable_encryption(bool create_new_db) {
+
+	fil_space_t*	space = fil_space_get(TRX_SYS_SPACE);
+	dberr_t		err = DB_SUCCESS;
+
+	if (create_new_db && srv_sys_tablespace_encrypt) {
+
+		space->flags |= FSP_FLAGS_MASK_ENCRYPTION;
+		srv_sys_space.set_flags(space->flags);
+
+		err = fil_set_encryption(space->id,
+					 Encryption::AES,
+					 NULL,
+					 NULL);
+		ut_ad(err == DB_SUCCESS);
```

Comment:
Is this required? I.e. wouldn't the release build give an fitting error up in the callstack?

Reply:
The assertion is superficial. Only when fil_space_t of tablespace is not found, it would fail. and this cannot happen for hardcoded sys / temp tablespace.

The assert is debug build only. If you wish, I can remove this.

Reply:
Thanks, then it's OK as-is

Reply:
Ok